src/abstractdataframe/abstractdataframe.jl

```diff
@@ -65,22 +66,29 @@ abstract type AbstractDataFrame end
     names(df::AbstractDataFrame)
     names(df::AbstractDataFrame, cols)
 
-    Return a `Vector{Symbol}` of names of columns contained in `df`.
+    Return a freshly allocated `Vector{String}` of names of columns contained in `df`.
 
-    If a `cols` column selector is passed then restrict returned
-    column names to those matching the selector
+    If a `cols` column selector ($COLUMN_INDICATOR, $COLUMNS_INDICATOR)
```

Comment:
The concatenation of the two may be a bit weird because of the "or integer" in the middle. Maybe separate them with a ";" (here and elsewhere)?

Also, inserting this long list in the middle of a sentence will be hard to read. Probably better add this at the end of the paragraph in a separate sentence like "`cols` can be any column selector (...)." Same in other places (including argument lists).

Reply:
OK - I will change this